examples/pygeosxExamples/hydaulicFractureWithMonitor/hydraulicFractureWithMonitor.py

```diff
@@ -0,0 +1,57 @@
+
+import sys
+from mpi4py import MPI
+import pygeosx
+from pygeosx_tools import wrapper, xml
+import matplotlib.pyplot as plt
+
+
+# Get the MPI rank
+comm = MPI.COMM_WORLD
+rank = comm.Get_rank()
+
+
+def run_problem():
+    """
+    Run the GEOSX problem
+    """
+    # Initialize the code
+    xml.apply_xml_preprocessor()
+    problem = pygeosx.initialize(rank, sys.argv)
+
+    # Apply initial conditions
+    pygeosx.apply_initial_conditions()
+
+    # Rather than specifying the wrapper paths explicitly,
+    # search for them using a set of filters
+    fracture_location_key = wrapper.get_matching_wrapper_path(problem, ['Fracture', 'elementCenter'])
+    fracture_aperture_key = wrapper.get_matching_wrapper_path(problem, ['Fracture', 'effectiveAperture'])
+
+    # Note: we will be plotting our results.
+    # This will modify the fonts so that they a bit easier to read
+    plot_font = {'weight': 'normal', 'size': 8}
+    plt.rc('font', **plot_font)
+
+    # Setup values to record
+    records = {fracture_location_key: {'label': 'Fracture Extents (m)',
+                                       'scale': 1.0,
+                                       'history': [],
+                                       'fhandle': plt.figure()},
```

Comment:
No, we shouldn't need a graphical back-end.  Before you import pyplot you can use a non-graphical one:

```python
import matplotlib
matplotlib.use('TkAgg')
import matplotlib.pyplot as plt
```


Also, if we do keep this functionality it would probably be worth turning this into a class to help end-users